### PR TITLE
python312Packages.diagrams-as-code: init at 0.0.4

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

Fixes:
- #331981

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/diagrams-as-code/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  pythonImportsCheck = [ "diagrams_as_code" ];
+
+  doCheck = false; # no tests
+
+  meta = {
+    description = "Declarative configurations using YAML for drawing cloud system architectures";
+    homepage = "https://github.com/dmytrostriletskyi/diagrams-as-code";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ sigmanificient ];
```
Comment (natsukium) [inline pkgs/development/python-modules/diagrams-as-code/default.nix:44]:
```suggestion
    maintainers = with lib.maintainers; [ sigmanificient ];
    mainProgram = "diagrams-as-code";
```
https://github.com/dmytrostriletskyi/diagrams-as-code/blob/f7682bbf8cd82ea250aad6b6779fcdde4372a7d8/setup.py#L42-L46

**Review by natsukium — COMMENTED** [review]
I think this is an application, but is it also used as a library?

**Comment by Sigmanificient** [comment]
tested with one of the examples

image

File: pkgs/development/python-modules/diagrams-as-code/default.nix (hunk 1 above)
Comment (Sigmanificient) [reply pkgs/development/python-modules/diagrams-as-code/default.nix:44]: Moved it to `pkgs/by-name`

**Comment by natsukium** [comment]
> tested with one of the examples

How about adding it to `passthru.tests`?

**Comment by Sigmanificient** [comment]
Would this do?
```nix
{
  stdenv,
  diagrams-as-code,
  lib
}:
stdenv.mkDerivation {
  name = "${diagrams-as-code.pname}-test";

  inherit (diagrams-as-code) src version meta;

  buildPhase = ''
    runHook preBuild

    ${lib.getExe diagrams-as-code} examples/all-fields.yaml

    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall

    cp web-services-architecture-aws.jpg $out

    runHook postInstall
  '';
}
```

**Comment by natsukium** [comment]
Great! But `runCommand` makes tests more simple.
https://github.com/NixOS/nixpkgs/tree/master/pkgs#writing-inline-package-tests

**Comment by Sigmanificient** [comment]
How do I retrieve `examples/all-fields.yaml` within `runCommand`?

File: pkgs/development/python-modules/diagrams-as-code/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/diagrams-as-code/default.nix:1]: Could you remove this file?

**Review by natsukium — COMMENTED** [review]
Please change the commit message from `python312Packages.diagrams-as-code: init at 0.0.4` to `diagrams-as-code: init at 0.0.4`.

**Pull request merged**